src/Infrastructure/Rokalo.Infrastructure.Db.Users/Repositories/UserRepository.cs

```diff
@@ -0,0 +1,40 @@
+﻿namespace Rokalo.Infrastructure.Db.Users.Repositories
+{
+    using Microsoft.EntityFrameworkCore;
+    using Rokalo.Application.Contracts;
+    using Rokalo.Domain;
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    internal sealed class UserRepository : IUserRepository
+    {
+        private readonly DbSet<User> users;
+
+        public UserRepository(UsersDbContext context)
+        {
+            this.users = context.Set<User>();
+        }
+        public void Create(User user)
+        {
+            this.users.Add(user);
+        }
+
+        public void Delete(User user)
+        {
+            this.users.Remove(user);
+        }
+
+        public async Task<User> GetByIdAsync(Guid id, CancellationToken cancellationToken)
+        {
+            return await this.users.FindAsync(new object[] {id}, cancellationToken) ?? throw new ApplicationException("Unable to find that user.");
```

Comment:
`namespace Company.Shorts.Blocks.Common.Exceptions
{
    using System;
    using System.Collections.Generic;

    public class ValidationException : ApplicationException
    {
        public ValidationException(Dictionary<string, string[]> errors)
        {
            Title = "One or more validation errors occurred.";
            Detail = "Check the errors for more details";
            Errors = errors;
        }

        public ValidationException(string error)
        {
            Title = "One or more validation errors occurred.";
            Detail = "Check the errors for more details";
            Errors = new Dictionary<string, string[]>()
            {
                { "General", new string[] { error } }
            };
        }

        public string Title { get; }

        public string Detail { get; }

        public Dictionary<string, string[]> Errors { get; }
    }
}
`